### ISSUE TITLE
Lingering metrics tags after delete

### ISSUE THREAD

**Issue by davidvgalbraith** [kairosdb/kairosdb#111]
Kairos! I'm back again with another strange bug. Running 0.9.4, the current time in UNIX milliseconds is about 1413613069972. So let's say I import a metric with this timestamp:

```
curl -XPOST -d '[{
"name": "spooky",
"timestamp": 1413613069972,
"value": 1,  
"tags": {"host": "server2"}}]' http://localhost:8080/api/v1/datapoints
```

And make sure it got in there ok:

```
curl -XPOST -d '{
"start_absolute": 0, 
"metrics": [{"name": "spooky"}]}' http://localhost:8080/api/v1/datapoints/query

{"queries":[{
"sample_size":1,
"results":[{
"name":"spooky",
"group_by":[{"name":"type","type":"number"}],
"tags":{"host":["server2"]},
"values":[[1413613069972,1]]}]}]}
```

Ok looking good but I don't like that spooky point so let's delete it:

```
curl -XPOST -d '{
"start_absolute": 0, 
"metrics": [{"name": "spooky"}]}' http://localhost:8080/api/v1/datapoints/delete
```

And make sure it got deleted ok:

```
curl -XPOST -d '{
"start_absolute": 0, 
"metrics": [{"name": "spooky"}]}' http://localhost:8080/api/v1/datapoints/query

{"queries":[{
"sample_size":0,
"results":[{"name":"spooky","tags":{},"values":[]}]}]}
```

Ok great but plot twist time let's query for spooky's tags:

```
curl -XPOST -d '{
"start_absolute": 0, 
"metrics": [{"name": "spooky"}]}' http://localhost:8080/api/v1/datapoints/query/tags

{"queries":[{"results":[{
"name":"spooky",
"tags":{"host":["server2"]},
"values":[]}]}]}
```

Jeepers! Zombie tags! How did those get on there? Let's try something crazy to get rid of them:

```
curl -XPOST -d '{
"start_absolute": 0, 
"end_absolute": 1513613069972,
"metrics": [{"name": "spooky"}]}' http://localhost:8080/api/v1/datapoints/delete

curl -XPOST -d '{
"start_absolute": 0, 
"metrics": [{"name": "spooky"}]}' http://localhost:8080/api/v1/datapoints/query/tags

{"queries":[{
"results":[
{"name":"spooky",
"tags":{},"values":[]}]}]}
```

So basically if you check out my second delete there you'll notice that the difference is I put an explicit end_absolute on there, and that end_absolute is 100 billion milliseconds in the future. So it seems like whatever you're getting the tags from in the tags query has some artifacts from the future that don't get erased when just doing delete up to the present moment, even if all the metric values get deleted.
